src/core/interfaces.ts

```diff
@@ -43,3 +43,7 @@ export interface INetworkConfig {
     gasPriceModifier: number;
     chainID: string;
 }
+
+export interface IGasLimitEstimator {
+    estimateGasLimit(transaction: Transaction): Promise<bigint>;
```

Comment:
Idea: maybe pass `{ transaction: Transaction }` (options), so that we'll avoid a possible breaking change when more parameters are required (e.g. additional transaction context necessary for _**some exotic**_ estimators).